pulser-simulation/pulser_simulation/hamiltonian.py

```diff
@@ -302,12 +307,18 @@ def add_noise(
                 ch_amp_fluctuation = max(
                     0, np.random.normal(1.0, self.config.amp_sigma)
                 )
+                ch_det_fluctuation = (
+                    np.random.normal(0.0, self.config.detuning_sigma)
+                    if self.config.detuning_sigma
+                    else 0.0
+                )
```

Comment:
nit: Since `detuning_sigma` is a float, I think this would suffice
```suggestion
                ch_det_fluctuation = np.random.normal(0.0, self.config.detuning_sigma)
```